robot/impl/utils.go

```diff
@@ -24,11 +24,19 @@ func setupLocalRobot(
 	r, err := New(ctx, cfg, logger, WithViamHomeDir(t.TempDir()))
 	test.That(t, err, test.ShouldBeNil)
 	t.Cleanup(func() {
-		test.That(t, r.Close(ctx), test.ShouldBeNil)
-		// Wait for reconfigureWorkers here because localRobot.Close does not.
-		lRobot, ok := r.(*localRobot)
-		test.That(t, ok, test.ShouldBeTrue)
-		lRobot.reconfigureWorkers.Wait()
+		test.That(t, r.CloseWait(ctx), test.ShouldBeNil)
 	})
 	return r
 }
+
+// SetupLocalRobot create a local robot from a given config and ensure it is shutdown
+// after the test is complete. This includes waiting for any reconfiguration workers to
+// finish.
+func SetupLocalRobot(
+	t *testing.T,
+	ctx context.Context,
+	cfg *config.Config,
+	logger logging.Logger,
+) robot.LocalRobot {
+	return setupLocalRobot(t, ctx, cfg, logger)
+}
```

Comment:
exposing for the benefit of native component tests

Reply:
not sure if we need this for the component tests